map_model/src/traversable.rs

```diff
@@ -212,32 +212,68 @@ impl Traversable {
         constraints: PathConstraints,
         map: &Map,
     ) -> Speed {
-        let base = match self {
-            Traversable::Lane(l) => {
-                let road = map.get_parent(*l);
-                let percent_incline = if road.dir(*l) == Direction::Fwd {
-                    road.percent_incline
-                } else {
-                    -road.percent_incline
-                };
-
-                if constraints == PathConstraints::Bike {
-                    // We assume every bike has a max_speed defined.
-                    bike_speed_on_incline(max_speed_on_flat_ground.unwrap(), percent_incline)
-                } else if constraints == PathConstraints::Pedestrian {
-                    // We assume every pedestrian has a max_speed defined.
-                    walking_speed_on_incline(max_speed_on_flat_ground.unwrap(), percent_incline)
-                } else {
-                    // Incline doesn't affect cars, buses, or trains
-                    road.speed_limit
-                }
-            }
-            // TODO Ignore elevation on turns?
-            Traversable::Turn(t) => map
-                .get_parent(t.src)
-                .speed_limit
-                .min(map.get_parent(t.dst).speed_limit),
+        match self {
+            Traversable::Lane(l) => Traversable::max_speed_along_road(
+                map.get_l(*l).get_directed_parent(map),
+                max_speed_on_flat_ground,
+                constraints,
+                map,
+            ),
+            Traversable::Turn(t) => Traversable::max_speed_along_movement(
+                t.to_movement(map),
+                max_speed_on_flat_ground,
+                constraints,
+                map,
+            ),
+        }
+    }
+
+    /// The single definitive place to determine how fast somebody could go along a single road.
+    /// This should be used for pathfinding and simulation.
+    pub fn max_speed_along_road(
```

Comment:
Maybe it's worth making `TraversableV2` that's either `DirectedRoadID` or `MovementID`. Not sure yet -- most uses are in the simulation layer. I'll revisit as I start to change that to natively understand road-based paths.